ethcore/src/encoded.rs

```diff
@@ -48,7 +48,7 @@ impl Header {
 	pub fn new(encoded: Vec<u8>) -> Self { Header(encoded) }
 
 	/// Upgrade this encoded view to a fully owned `Header` object.
-	pub fn decode(&self) -> FullHeader { ::rlp::decode(&self.0) }
+	pub fn decode(&self) -> FullHeader { ::rlp::decode(&self.0).unwrap_or(FullHeader::default()) } // REVIEW: is the default ok here? Useful? The real fix would be to make `decode` return a `Result` as well.
```

Comment:
It should definitely return `Result`. It it requires too many changes, let's replace it with `.expect` for now and let's create another issues to track this

Reply:
https://github.com/paritytech/parity/issues/8553